File: app/src/main/java/com/example/appbank/ui/login/LoginContract.java
package com.example.appbank.ui.login;

import com.example.appbank.data.remote.model.LoginRequest;
import com.example.appbank.data.remote.model.LoginResponse;

public interface LoginContract {

    interface LoginView{

        void showError();

        void navigateToList(LoginResponse loginResponse);

        void errorUsername(String message);

        void enabledButton(boolean b);

        void errorPassword(String s);

    }

    interface LoginPresenter{

        void login(LoginRequest loginRequest);

        void validUserData(String userName);

        boolean validPasswordData(String password);
    }
}

File: DesafioSantander/app/src/main/java/com/example/desafiosantander/data/repository/login/LoginContract.kt
package com.example.desafiosantander.data.repository.login

import com.example.desafiosantander.data.model.request.LoginRequest
import com.example.desafiosantander.data.model.response.LoginResponse
import io.reactivex.Flowable

interface LoginContract {

    fun login(loginRequest: LoginRequest): Flowable<LoginResponse>

}